torch/distributed/algorithms/ddp_comm_hooks/powerSGD_hook.py

```diff
@@ -162,38 +161,44 @@ def maybe_increase_iter(self, bucket):
 
 
 def powerSGD_hook(state: PowerSGDState, bucket) -> torch.futures.Future:
-    """
+    r"""
     This DDP communication hook implements the original PowerSGD gradient compression
     algorithm described in https://arxiv.org/abs/1905.13727.
     Once gradient tensors are aggregated across all workers, this hook applies
     compression as follows:
-    1) Views the input flattened 1D gradient tensor as two groups of per-parameter tensors:
-    high-rank tensors and vector-like rank-1 tensors (for biases).
-    2) Handles rank-1 tensors by allreducing them without compression:
-        2.1) Allocate contiguous memory for those rank-1 tensors,
-        and allreduces all the rank-1 tensors as a batch, without compression;
-        2.2) Copies the individual rank-1 tensors from the contiguous memory back to the input tensor.
-    3) Handles high-rank tensors by PowerSGD compression:
-        3.1) For each high-rank tensor M, creates two low-rank tensors P and Q for decomposing M,
+
+    1. Views the input flattened 1D gradient tensor as two groups of per-parameter tensors: high-rank tensors and vector-like rank-1 tensors (for biases).
+
+    2. Handles rank-1 tensors by allreducing them without compression:
+
+        2.1. Allocate contiguous memory for those rank-1 tensors, and allreduces all the rank-1 tensors as a batch, without compression;
+
+        2.2. Copies the individual rank-1 tensors from the contiguous memory back to the input tensor.
+
+    3. Handles high-rank tensors by PowerSGD compression:
+
+        3.1. For each high-rank tensor M, creates two low-rank tensors P and Q for decomposing M,
         such that M = PQ^T, where Q is initialized from a standard normal distribution and orthogonalized;
-        3.2) Computes each P in Ps, which is equal to MQ;
-        3.3) Allreduces Ps as a batch;
-        3.4) Orthogonalizes each P in Ps;
-        3.5) Computes each Q in Qs, which is approximately equal to M^TP;
-        3.6) Allreduces Qs as a batch;
-        3.7) Computes each M among all the high-rank tensors, which is approximately equal to PQ^T.
-
-    Note that this communication hook enforces vanilla allreduce for the first `state.start_powerSGD_iter` iterations.
+
+        3.2. Computes each P in Ps, which is equal to MQ;
+
+        3.3. Allreduces Ps as a batch;
+
+        3.4. Orthogonalizes each P in Ps;
+
+        3.5. Computes each Q in Qs, which is approximately equal to M^TP;
+
+        3.6. Allreduces Qs as a batch;
+
+        3.7. Computes each M among all the high-rank tensors, which is approximately equal to PQ^T.
```

Comment:
I feel like the #.# steps here are maybe too detailed for the end-user documentation.
Maybe you could refer the reader to the paper for details on the algorithm.
The content here also duplicates inline comments, and could lead to inconsistencies.

Maybe something higher level could be
```
2. Gradient tensors are reshaped into a matrix M, where the rows and columns correspond to input- and output dimensions of the layer. 
3. After adding accumulated compression errors from before, this matrix is approximated as low-rank matrix that can be cheaply averaged between workers.
```

Reply:
A bit more details are fine, because the pseudocode in the paper cannot cover all the details, such as handling rank-1 tensors, and Ps and Qs are allreduced in batches, etc.

Reply:
Ok, makes sense. I guess there are two targets for this doc: people who want to work with this code and people who just want to use it. To satisfy both, maybe you could should start with a simple and short overview of what happens (along the lines of the suggestion here or even shorter), and then provide details afterwards?